test/Lifecycle-test.ts

```diff
@@ -628,16 +647,132 @@ describe("Lifecycle", () => {
             it("should create new matrix client with credentials", async () => {
                 expect(await setLoggedIn(credentials)).toEqual(mockClient);
 
-                expect(MatrixClientPeg.replaceUsingCreds).toHaveBeenCalledWith({
-                    userId,
-                    accessToken,
-                    homeserverUrl,
-                    identityServerUrl,
-                    deviceId,
-                    freshLogin: true,
-                    guest: false,
-                    pickleKey: expect.any(String),
+                expect(MatrixClientPeg.replaceUsingCreds).toHaveBeenCalledWith(
+                    {
+                        userId,
+                        accessToken,
+                        homeserverUrl,
+                        identityServerUrl,
+                        deviceId,
+                        freshLogin: true,
+                        guest: false,
+                        pickleKey: expect.any(String),
+                    },
+                    undefined,
+                );
+            });
+        });
+
+        describe("when authenticated via OIDC native flow", () => {
+            const clientId = "test-client-id";
+            const issuer = "https://auth.com/";
+
+            const delegatedAuthConfig = makeDelegatedAuthConfig(issuer);
+            const idTokenClaims = {
+                aud: "123",
+                iss: issuer,
+                sub: "123",
+                exp: 123,
+                iat: 456,
+            };
+
+            beforeAll(() => {
+                fetchMock.get(
+                    `${delegatedAuthConfig.issuer}.well-known/openid-configuration`,
+                    delegatedAuthConfig.metadata,
+                );
+                fetchMock.get(`${delegatedAuthConfig.issuer}jwks`, {
+                    status: 200,
+                    headers: {
+                        "Content-Type": "application/json",
+                    },
+                    keys: [],
+                });
+            });
+
+            beforeEach(() => {
+                // mock oidc config for oidc client initialisation
+                mockClient.getClientWellKnown.mockReturnValue({
+                    [MatrixJs.M_AUTHENTICATION.stable!]: {
```

Comment:
given that this is a specced identifier rather than anything internal, I would use the literal value in test code:

```suggestion
                    "m.authentication": {
```

Doing so has three advantages imho:
 * It guards against implementation errors in `M_AUTHENTICATION`.
 * If someone decides to change the definition of `M_AUTHENTICATION`, the fact that the test will continue to use the old value means that we will be forced to think about how the new definition interacts with existing deployments (which of course will not change).
 * I find it easier to read, though this is subjective I guess.

YMMV, I don't feel strongly